### PR TITLE
feat: add functions isValidLandlinePhone() and isValidMobilePhone()

### PR DESCRIPTION
**Pull request by silvioprog**
Issued at https://github.com/brazilian-utils/brazilian-utils/issues/97.

### CODE REVIEW

File: src/utilities/phone/index.ts [hunk 1]
```diff
@@ -18,20 +18,56 @@ export function isValidDDD(phone: string): boolean {
...
 
 export function isValidFirstNumber(phone: string): boolean {
   return phone.length === PHONE_MIN_LENGTH
-    ? LANDLINE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)))
-    : MOBILE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
+    ? isValidLandlinePhoneFirstNumber(phone)
+    : isValidMobilePhoneFirstNumber(phone);
+}
+
+function tryPhoneDigits(phone: string): { isValidDigits: boolean; digits: string } {
```
Comment (hyanmandian) [inline src/utilities/phone/index.ts:47]: Maybe `parsePhoneDigits` is a better function name.

File: src/utilities/phone/index.ts [hunk 2]
```diff
@@ -18,20 +18,56 @@ export function isValidDDD(phone: string): boolean {
...
+  const { isValidDigits, digits } = tryPhoneDigits(phone);
+
+  if (!isValidDigits) return false;
+
+  return isValidLandlinePhoneLength(digits) && isValidLandlinePhoneFirstNumber(digits) && isValidDDD(digits);
 }
 
 export function isValid(phone: string): boolean {
-  if (!phone || typeof phone !== 'string') return false;
+  const { isValidDigits, digits } = tryPhoneDigits(phone);
```
Comment (hyanmandian) [inline src/utilities/phone/index.ts:68]: I think that you could rewrite to something like this:

```
export function isValid(phone: string): boolean {
  return isValidLandlinePhone(phone) || isValidMobilePhone(phone);
}
```

What do you think about it?

File: src/utilities/phone/index.ts (hunk 1 above)
Comment (silvioprog) [reply src/utilities/phone/index.ts:47]: Indeed. I'm going to change it.

File: src/utilities/phone/index.ts (hunk 2 above)
Comment (silvioprog) [reply src/utilities/phone/index.ts:68]: Hm.... It was to avoid to call `tryPhoneDigits()` two times, however, I'm going to change as suggested.

**Comment by silvioprog** [comment]
All the changes were applied as suggested. Thank you very must for reviewing! :-)

**Comment by hyanmandian** [comment]
Hey @silvioprog, can you check the coverage report? The CI was failed.

**Comment by silvioprog** [comment]
> Hey @silvioprog, can you check the coverage report? The CI was failed.

So, those warnings were emitted after applying the changes you suggested. Let me know if you prefer the previous one, so I can revert it.

**Comment by hyanmandian** [comment]
> > Hey @silvioprog, can you check the coverage report? The CI was failed.
> 
> So, those warnings were emitted after applying the changes you suggested. Let me know if you prefer the [previous one](https://github.com/brazilian-utils/brazilian-utils/pull/98/commits/e8fe8bd8aef422d0cc09f09dce232e0352540842), so I can revert to it.

I think when you applying the changes that I suggested, the coverage decreased. Maybe checking the code coverage you can found which patch wasn't coverage by tests.

**Comment by silvioprog** [comment]
Now all Codecov warnings were fixed. :-)

**Comment by hyanmandian** [comment]
Done! I published and you can use in `@brazilian-utils/brazilian-utils@1.0.0-rc.9`. Awesome job @silvioprog !

**Comment by silvioprog** [comment]
Thank you very much for merging! :+1:

**Pull request merged**